lib/ansible/modules/cloud/amazon/ec2_vpc_vgw.py

```diff
@@ -202,9 +206,10 @@ def detach_vgw(client, module, vpn_gateway_id, vpc_id=None):
 def create_vgw(client, module):
     params = dict()
     params['Type'] = module.params.get('type')
+    params['AmazonSideAsn'] = module.params.get('asn')
```

Comment:
Can you make this conditional, so it's not added as `None` because I think boto3 will raise a validation error if the int param is a NoneType. Something like:

```
if module.params.get('asn'):
    params['AmazonSideAsn'] = module.params.get('asn')
```

Should do the trick.